### PR TITLE
feat: add gitrepository types

### PR DESCRIPTION
**Pull request by kinsolee**
# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [`spec` PR link](https://github.com/katanomi/spec) included
- [ ] Follows the [commit message standard](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md#commits)
- [ ] Meets the [contributing guidelines](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md) (including
  functionality, content, code)
- [ ] Test cases with documentation and functionality works as expected using current and related github repos (MUST deploy and check)
- [ ] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

### CODE REVIEW

File: apis/coderepository/v1alpha1/gitrepository_types.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+const (
+	// GitRepositoryURLKey is the key for the code repository URL
+	GitRepositoryURLKey = "integrations.katanomi.dev/gitRepositoryURL"
+)
+
+// GitRepositoryAddress contains the address of the git repository
+type GitRepositoryAddress struct {
```
Comment (danielfbm) [inline apis/coderepository/v1alpha1/gitrepository_types.go:27]: 这个API是要在哪里使用呢？为啥跟我们已有的标准不一样的？（spec, status)

Comment (danielfbm) [reply apis/coderepository/v1alpha1/gitrepository_types.go:27]: 感觉很奇怪，为啥要有一个GitRepositoryAddress 的struct，然后里面主要信息是一个url数组？

**Review by danielfbm — CHANGES_REQUESTED** [review]
留了评论

**Comment by kinsolee** [comment]
迁移到integrations仓库：https://github.com/katanomi/integrations/pull/311/files#diff-556a7955f5282ce3c07c9b077aa28466db987c48e3764ba6b6705eeeb9c9c352

**Pull request closed without merging**